lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -4193,6 +4195,26 @@ ParseResult FIRCircuitParser::parseParameterList(ArrayAttr &resultParameters) {
   return success();
 }
 
+/// class ::= 'class' id ':' info? INDENT portlist simple_stmt_block DEDENT
+ParseResult FIRCircuitParser::parseClass(CircuitOp circuit, unsigned indent) {
+  StringAttr name;
+  SmallVector<PortInfo, 8> portList;
+  SmallVector<SMLoc> portLocs;
+  LocWithInfo info(getToken().getLoc(), this);
+  consumeToken(FIRToken::kw_class);
+  if (parseId(name, "expected class name") ||
+      parseToken(FIRToken::colon, "expected ':' in class definition") ||
+      info.parseOptionalInfo() || parsePortList(portList, portLocs, indent))
```

Comment:
Should this be checking the ports for validity (only property-type)?

Verifier seems reasonable but checking in parser (or invoking verification logic somehow?) to produce nice diagnostics can be nice re:better error messages.

Reply:
Came to suggest an error test and giving these non-property ports was my first thing to try :smile: .

Reply:
added two simple checks below, and an error test for each. We still need to reject hardware ops under a class, but if it's okay, I will save that for a separate PR (it's not properly verified in the IR yet, either).

Reply:
Totally, sounds great! :+1: